### ISSUE TITLE
Unable to Unlock previously secured KL17

### ISSUE THREAD

**Issue by RomanGassmann** [podonoghue/usbdm-eclipse-makefiles-build#28]
Hi Peter,

I just wanted to let you know about a problem I recently had with the latest version of the UsbdmFlashProgrammer.
I had to build the UsbdmFlashProgrammer myself due to a system change (to Debian 12) on my end.
So I pulled the latest version (14.12.2023) and updated the dependencies.
Everything went well and I was happy to have the latest version built and running in no time (including flashing devices).
But yesterday my luck ran out when I tried to flash a device that was previously flashed with the secure option.
The UsbdmFlashProgrammer stucks and always reports "TCL error", which is ok when programming a device, but not with Masserase. Unfortunately, I haven't had time (yet) to investigate this further, but looking at the repository history I saw a 
commit message "Changes to the rest sequence for the MK series (with TCL)". So I wonder if this could be related?

I was able to unlock my device by running KinetisUnlock and this proves (at least I think so) that the system should work.
Over the next few days I will investigate this further and keep you posted.
If you already have an idea where the problem lies, I would very much appreciate any 
information from you and would also carry out appropriate tests on my part.

Best regards,
Roman

**Comment by podonoghue** [comment]
Hi Roman,

I'm sorry I am experimenting with the Kinetis flash programmer to allow the reset and connection to be more customised in the TCL scripts.
This has only be done for the MKxx devices and only a small amount of testing on a MK22F device using a particular version of the BDM (one with target power control).

This was to cope with some problems with connecting to devices that have the reset line pulsing due to watch-dog or illegal-instruction traps.
Basically programming may be broken on other devices or with a different programmer version.

I suggest you revert to a version before that commit.
It is not worth your time testing as it is a work in progress.

bye

**Comment by podonoghue** [comment]
Hi Roman,
Updated TCL scripts so they should now work on KL devices when using a BDM without target Vdd control.
If there is still a problem please reply with details of the BDM and target.

There is no urgency in this.

bye

**Comment by RomanGassmann** [comment]
Hi Peter,
thanks for your info on this and sorry for my late response (I was quiet busy the last months).
I just checked the newest release, and I can confirm, its working a gain - without any TLC errors. Thanks. 
One thing I noticed though, in previous versions it was possible to flash new software even to secured devices, when chosen "EraseMass" as Erase Mode.
What do you think, would it be possible to actually unlock a secured device previously to flash under some circumstances like the chosen Erase Mode - this with regard to your recent expansion of the software to support other devices as well.
If so I'd be happy to try different things myself - in that case I would be glad if you could push me into the right direction on where to fiddle in order to be as efficient as possible here.
Thanks and best,
Roman
